test/echo_test.exs

```diff
@@ -2,7 +2,62 @@ defmodule EchoTest do
   use ExUnit.Case
   doctest Echo
 
-  test "greets the world" do
-    assert Echo.hello() == :world
+  test "we can run the echo process" do
+    {:ok , pid} = Echo.start(self())
+    assert(is_pid(pid))
+  end
+
+  test "should get the result when sending :ping" do
+    {:ok, pid} = Echo.start(self())
+
+    send(pid, :ping)
+
+    receive do
```

Comment:
This is always cool to invent the wheel, but people have been there already. [`assert_receive/3`](https://hexdocs.pm/ex_unit/ExUnit.Assertions.html#assert_receive/3).